spec/components/s3_inventory_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+require "s3_inventory"
+require "file_store/s3_store"
+
+describe "S3Inventory" do
+  let(:store) { FileStore::S3Store.new }
+  let(:inventory) { store.inventory }
+  let(:csv_filename) { File.new("#{Rails.root}/spec/fixtures/csv/s3_inventory.csv") }
+
+  before do
+    SiteSetting.enable_s3_uploads = true
+    SiteSetting.s3_access_key_id = "abc"
+    SiteSetting.s3_secret_access_key = "def"
+    SiteSetting.s3_inventory_bucket = "ghi"
+    SiteSetting.enable_s3_inventory = true
+
+    s3_helper = store.inventory.instance_variable_get(:@s3_helper)
```

Comment:
Maybe we should add `s3_helper` to `attr_reader` if the instance variable needs to be accessed?

Reply:
Yes, I also think we need few more refactoring here.